google/cloud/aiplatform/experimental/vertex_model/base.py

```diff
@@ -323,23 +343,24 @@ def p(*args, **kwargs):
 
         # Make remote predictions, regardless of training: create custom container
         if method.__self__.remote:
-            # Convert the predict input to a JSON input for the Endpoint resource
+            # Convert the predict input to a predict_payload input for the Endpoint resource
             data = []
             bound_args = inspect.signature(method).bind(*args, **kwargs)
 
             for parameter_name, parameter in bound_args.arguments.items():
                 if parameter_name == "data":
-                    data = parameter.tolist()
+                    data = obj.predict_input_to_predict_payload(parameter)
```

Comment:
I don't think we need the constrain the name this argument `data`. Currently we are bound to a single argument on predict because we cannot forward any of the additional arguments when calling the endpoint. I think we should constrain predict to single argument but not enforce that it is name be `data`. Also update the documentation to reflect.